internal/printer/printer_test.go

```diff
@@ -200,6 +200,71 @@ const name = "world";
   </body></html>`,
 			},
 		},
+		{
+			name: "React framework example",
+			source: `---
+// Component Imports
+import Counter from '../components/Counter.jsx'
+const someProps = {
+  count: 0,
+}
+
+// Full Astro Component Syntax:
+// https://docs.astro.build/core-concepts/astro-components/
+---
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta
+      name="viewport"
+      content="width=device-width"
+    />
+    <link rel="icon" type="image/x-icon" href="/favicon.ico" />
+    <style>
+      :global(:root) {
+        font-family: system-ui;
+        padding: 2em 0;
+      }
+      :global(.counter) {
+        display: grid;
+        grid-template-columns: repeat(3, minmax(0, 1fr));
+        place-items: center;
+        font-size: 2em;
+        margin-top: 2em;
+      }
+      :global(.children) {
+        display: grid;
+        place-items: center;
+        margin-bottom: 2em;
+      }
+    </style>
+  </head>
+  <body>
+    <main>
+      <Counter {...someProps} client:visible>
+        <h1>Hello React!</h1>
```

Comment:
This slot, along with the spread props, may be what’s hanging it up? Anyway this test just never completes, which matches what I was seeing from the npm version. Let’s take a look at it next week!